realm-flipper-plugin-device/src/RealmPlugin.tsx

```diff
@@ -1,86 +1,82 @@
-import React, { useEffect } from "react";
-import { addPlugin, Flipper } from "react-native-flipper";
-import Realm, { CanonicalObjectSchema } from "realm";
+import React, {useEffect, useRef} from 'react';
+import {addPlugin} from 'react-native-flipper';
+import type Realm from 'realm';
```

Comment:
Is there a reason to `import type` rather than just `import`?

Reply:
I figured this might be better purely for optimisation purposes from TypeScript point of view since Realm is used as a solely as a type definition import here. I imagine type imports do not get considered in JS code at all whereas full-on imports might be? This is less relevant in the device library and is more of a consequence of me trying to make it explicit that Realm is used only for types as a dev dependency in the desktop plugin.

Reply:
It piqued my curiosity, so I did some googling:
https://github.com/microsoft/TypeScript/issues/39861#issuecomment-668131921
Seems it's not really necessary, and is more of a stylistic choice.

Reply:
that's quite interesting. I feel like I disagree with best style being never using typed imports though... It seems valuable to enforce not using those imports as values for example, i.e. in case of flipper desktop plugin it makes it explicit there's no actual Realm or active Realm objects that exist, just abstractions of their types. I think I'd like to keep it in the desktop plugin just for that reason but it's probably confusing in the device plugin so I could remove that?

I honestly thought it would be deeper than that as I was working with Lua and a Language Server which also introduces a type system for Lua, this distinction was more drastic because there is nothing that ends up actually being imported with type imports (it was basically just a comment that added a path for the language server to discover definitions) whereas traditional imports would actually run code import files etc. etc.. But yeah, interesting stuff.